hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,256 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "edn",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  bus_host: "none",
+  param_list: [
+    { name: "NumEndPoints",
+      desc: "Number of end points to be supported by this block",
+      type: "int",
+      default: "4",
+      local: "true"
+    },
+    { name: "EndPointBusWidth0",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 0",
+      type: "int",
+      default: "64",
+      local: "true"
+    },
+    { name: "EndPointBusWidth1",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 1",
+      type: "int",
+      default: "32",
+      local: "true"
+    },
+    { name: "EndPointBusWidth2",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 2",
+      type: "int",
+      default: "16",
+      local: "true"
+    },
+    { name: "EndPointBusWidth3",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 3",
+      type: "int",
+      default: "8",
+      local: "true"
+    },
+  ],
+  interrupt_list: [
+    { name: "edn_cmd_req_done"
+      desc: "Asserted when a software CSRNG request has completed."}
+    { name: "edn_fifo_err"
+      desc: "Asserted when a FIFO error occurs."}
+  ],
+  inter_signal_list: [
+    { struct:  "edn",  // edn_req_t, edn_rsp_t
+      type:    "req_rsp",
+      name:    "edn_cmd",
+      act:     "rsp",
+      width:   "4",
+      desc:    '''
+               The collection of peripheral ports supported by edn. The width (4)
+               indicates the number of peripheral ports on a single instance.
+               Due to limitations in the parametrization of top-level interconnects
+               this value is not currently parameterizable.  However, the number
+               of peripheral ports may change in a future revision.
+               ''',
+      package: "edn_pkg",
+    }
+  ],
+
+  regwidth: "32",
+  registers: [
+    { name: "REGEN",
+      desc: "Register write enable for all control registers",
+      swaccess: "rw1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: '''
+                When true, the BOOT_REQ_DIS and BCAST_MODE_DIS fields within the CTRL registers
+                can be modified. When false, this field read-only. Defaults true, write one to clear.
+                Note that this needs to be cleared after initial configuration at boot in order to
+                lock in the listed register settings.
+                '''
+          resval: 1
+        },
+      ]
+    },
+    { name: "CTRL",
+      desc: "EDN control register",
+      swaccess: "rw",
+      hwaccess: "hro",
```

Comment:
The answer might just be "Yes!", but is this the right set of flags? I would have expected `CMD_FIFO_RST` to need something more like `rw1c`?

Reply:
I am assuming these flags are on the CTRL register. If so, these flags give the most flexibility. Clear and set flags have there place, but not for this control reg.